### DISCUSSION TITLE
How to add string parameter without casting?

### DISCUSSION THREAD

**Discussion by isgj** [go-jet/jet#176] [Q&A] [answered]
Now if I want to add a string parameter without casting I do
```go
Raw("#p", RawArgs{"#p": "some value"})
```
Is there an easier helper function?

I want the resulting query to have `$1` (without `::text`) and the value to be part of the parameters

**Comment by go-jet** [comment, marked as answer]
@isgj Yeah, that would be the easier way to add query parameter without explicit cast. 
Why do you need such a parameter? What type is `$1` parameter?

**Reply by isgj** [reply]
`$1` is of type `uuid` and I've got the value as string (some times). I don't like to parse it as uuid and then just pass it to `Uuid` which will call `String` to get the string. I saw `Uuid` is just a wrapper to `internal.String`, maybe you can export the `internal.String` just to have a StringExpression without `::text` casting

**Reply by go-jet** [reply]
If that's the case, I would say it works as intended. It is forcing you to use exact column uuid type rather then string. If the string is not a valid uuid, you'll get a clean error notification early, and you wouldn't have to parse database error message to find out what is wrong.
Consider using UUID type as request struct field, that way you could automate uuid parse during request unmarshal.

**Reply by isgj** [reply]
> If the string is not a valid uuid, you'll get a clean error notification early

Well, `UUID` accepts a `Stringer`, the db error will be the same for non valid uuid, it won't be catched by `UUID`.

> Consider using UUID type as request struct field

It's not from the request and the structure comes from a 3rd party lib.

Now I implement `Stringer` like (I find this better):

```go
type UuidString string

func (u UuidString) String() string { return string(u) }

// and use it
....UUID(UuidString(value))
```

**Reply by go-jet** [reply]
`Stringer` is the only interface shared between most popular Go uuid implemenation.
`UuidString` will work as well.
